ilastik/applets/batchProcessing/batchProcessingApplet.py

```diff
@@ -51,219 +53,99 @@ def dataSerializers(self):
 
     def parse_known_cmdline_args(self, cmdline_args):
         # We use the same parser as the DataSelectionApplet
-        role_names = self.dataSelectionApplet.topLevelOperator.DatasetRoles.value
-        parsed_args, unused_args = DataSelectionApplet.parse_known_cmdline_args(
-            cmdline_args, role_names)
+        parsed_args, unused_args = DataSelectionApplet.parse_known_cmdline_args(cmdline_args, self.role_names)
         return parsed_args, unused_args
 
     def run_export_from_parsed_args(self, parsed_args):
         """
         Run the export for each dataset listed in parsed_args (we use the same parser as DataSelectionApplet).
         """
-        role_names = self.dataSelectionApplet.topLevelOperator.DatasetRoles.value
-        role_path_dict = self.dataSelectionApplet.role_paths_from_parsed_args(
-            parsed_args, role_names)
+        role_path_dict = self.dataSelectionApplet.role_paths_from_parsed_args(parsed_args)
         return self.run_export(role_path_dict, parsed_args.input_axes, sequence_axis=parsed_args.stack_along)
 
-    def run_export(self, role_data_dict, input_axes=None, export_to_array=False, sequence_axis=None):
-        """
-        Run the export for each dataset listed in role_data_dict,
-        which must be a dict of {role_index : path-list} OR {role_index : DatasetInfo-list}
-
-        As shown above, you may pass either filepaths OR preconfigured DatasetInfo objects.
-        The latter is useful if you are batch processing data that already exists in memory as a numpy array.
-        (See DatasetInfo.preloaded_array for how to provide a numpy array instead of a filepath.)
+    def run_export(self, role_data_dict:Dict[int, List[str]], input_axes:str=None, export_to_array:bool=False, sequence_axis:str=None):
+        self.progressSignal(0)
+        batches = list(zip(*role_data_dict.values()))
+        try:
+            results = []
+            for batch_dataset_index, role_input_paths in enumerate(batches):
+                def emit_progress(dataset_percent):
+                    overall_progress = (batch_dataset_index + dataset_percent / 100.0) / len(batches)
```

Comment:
It looks like there is a late-binding problem here.

Reply:
I think I saw this bug during the last ilastik workshow when we export big file sequence progress bar resets to 0% at `100% / num_files` for every file instead of going to 100%

Reply:
Nice catch. Fixed =)